cmd/create/operatorroles/cmd.go

```diff
@@ -197,40 +208,253 @@ func run(cmd *cobra.Command, argv []string) error {
+		os.Exit(1)
+	}
+	// TODO: remove once AWS managed policies are in place
+	if managedPolicies && env == ocm.Production {
+		r.Reporter.Errorf("Managed policies are not supported in this environment")
+		os.Exit(1)
+	}
+	awsCreator, err := r.AWSClient.GetCreator()
+	if err != nil {
+		r.Reporter.Errorf("Unable to get IAM credentials: %v", err)
+		os.Exit(1)
+	}
+	operatorIAMRoleList := []*cmv1.OperatorIAMRole{}
+	for _, operator := range credRequests {
+		operatorIamRole, err := cmv1.NewOperatorIAMRole().
+			Name(operator.Name()).
+			Namespace(operator.Namespace()).
+			RoleARN(getOperatorRoleArn(operatorRolesPrefix, operator, awsCreator, path)).
+			Build()
+		if err != nil {
+			r.Reporter.Errorf("%s", err)
+			os.Exit(1)
+		}
+		operatorIAMRoleList = append(operatorIAMRoleList, operatorIamRole)
+	}
+	switch mode {
+	case aws.ModeAuto:
+		if !output.HasFlag() || r.Reporter.IsTerminal() {
+			r.Reporter.Infof("Creating roles using '%s'", r.Creator.ARN)
+		}
+		err = createRolesByPrefix(r, env,
+			operatorRolePolicyPrefix, permissionsBoundary,
+			defaultPolicyVersion, policies,
+			credRequests, managedPolicies,
+			path, operatorIAMRoleList,
+			oidcEndpointUrl)
+		if err != nil {
+			r.Reporter.Errorf("There was an error creating the operator roles: %s", err)
+			isThrottle := "false"
+			if strings.Contains(err.Error(), "Throttling") {
+				isThrottle = helper.True
+			}
+			r.OCMClient.LogEvent("ROSACreateOperatorRolesModeAuto", map[string]string{
+				ocm.OperatorRolesPrefix: operatorRolesPrefix,
+				ocm.Response:            ocm.Failure,
+				ocm.IsThrottle:          isThrottle,
+			})
+			os.Exit(1)
+		}
+		r.OCMClient.LogEvent("ROSACreateOperatorRolesModeAuto", map[string]string{
+			ocm.OperatorRolesPrefix: operatorRolesPrefix,
+			ocm.Response:            ocm.Success,
+		})
+	case aws.ModeManual:
+		commands, err := buildCommandsFromPrefix(r, env,
+			operatorRolePolicyPrefix, permissionsBoundary,
+			defaultPolicyVersion, policies,
+			credRequests, managedPolicies,
+			path, operatorIAMRoleList,
+			oidcEndpointUrl)
+		if err != nil {
+			r.Reporter.Errorf("There was an error building the list of resources: %s", err)
+			os.Exit(1)
+			r.OCMClient.LogEvent("ROSACreateOperatorRolesModeManual", map[string]string{
+				ocm.OperatorRolesPrefix: operatorRolesPrefix,
+				ocm.Response:            ocm.Failure,
+			})
+		}
+		if r.Reporter.IsTerminal() {
+			r.Reporter.Infof("All policy files saved to the current directory")
+			r.Reporter.Infof("Run the following commands to create the operator roles:\n")
+		}
+		r.OCMClient.LogEvent("ROSACreateOperatorRolesModeManual", map[string]string{
+			ocm.OperatorRolesPrefix: operatorRolesPrefix,
+		})
+		fmt.Println(commands)
+	default:
+		r.Reporter.Errorf("Invalid mode. Allowed values are %s", aws.Modes)
 		os.Exit(1)
 	}
+	return nil
+}
 
-	env, err := ocm.GetEnv()
+func getOperatorRoleArn(prefix string, operator *cmv1.STSOperator, creator *aws.Creator, path string) string {
```

Comment:
This function is very similar to `getOperatorRoleArn` in `cmd/create/cluster/cmd.go`.
Can we extract the function and reuse it?

Reply:
Will check in on that